src/main/java/org/broadinstitute/hellbender/utils/nio/PathLineIterator.java

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+/**
+ * Iterate through the lines of a Path. Works for anything you can point
+ * a Path to, such as a normal file or anything you have an NIO provider
+ * for (e.g. GCS).
+ */
+public class PathLineIterator implements AutoCloseable, Iterable<String> {
+  private final Stream<String> lines;
+
+  /**
+   * Returns an iterator so you can iterate over the lines in the text file like so:
+   * for (String line: new Utils.LineIterator(path)) {
+   *   // do something with the line
+   * }
+   *
+   * It's also closeable so you can close it when done, or use it in a try-with-resources
+   * to close it automatically.
+   *
+   * @param path path to a text file.
+   * @throws UserException if we cannot open the file for reading.
+   */
+  public PathLineIterator(Path path) throws UserException {
+    try {
+      lines = Files.lines(path);
```

Comment:
Maybe check for nonNull for path (e.g., `lines = Files.lines(Utils.nonNull(path, "path is null"));`

Reply:
Done.